### DISCUSSION TITLE
Github copilot is not working

### DISCUSSION THREAD

**Discussion by jianmincat** [community/community#127470] [GitHub Education]
### Hi everyone,

I got education license of github copilot, while I am stepping through the [experience-primer-copilot](https://github.com/Exp-Primer-Copilot-Cohort-1/experience-primer-copilot-jianmincat), I noticed the copilot suggestion is not functioning at all in Step 2. Can someone advise how should I proceed and resolve this issue. Many thanks.

**Comment by erick-hz** [comment]
Certainly! Here's a structured approach to troubleshoot and resolve the issue with GitHub Copilot not functioning properly in Step 2 of the experience-primer-copilot:

Check System Requirements:

Ensure that your system meets the minimum requirements for running GitHub Copilot. This includes hardware specifications and software dependencies.
Verify that your internet connection is stable and that there are no network issues preventing GitHub Copilot from accessing necessary resources.
Verify GitHub Copilot Activation:

Confirm that your GitHub Copilot license is properly activated and associated with your GitHub account.
Check for any notifications or prompts within GitHub or your IDE regarding the status of your GitHub Copilot license.
Update IDE and GitHub Copilot Extension:

Make sure that your Integrated Development Environment (IDE) is up to date with the latest version.
Update the GitHub Copilot extension/plugin within your IDE to the latest version available.
Restart your IDE after updating to ensure that the changes take effect.
Review IDE Settings and Configuration:

Double-check the settings and preferences within your IDE related to GitHub Copilot. Ensure that it's enabled and configured correctly.
Look for any conflicting extensions or plugins that might be interfering with GitHub Copilot's functionality. Disable them temporarily to see if it resolves the issue.
Clear Cache and Restart IDE:

Clear the cache and temporary files related to your IDE. This can help resolve any caching issues that might be affecting GitHub Copilot.
Restart your IDE after clearing the cache to allow it to rebuild necessary files and configurations.
Check GitHub Copilot Documentation and Support:

Refer to the official documentation and troubleshooting guides provided by GitHub Copilot for any specific issues or solutions related to Step 2 of the experience-primer-copilot.
If the problem persists, consider reaching out to GitHub support or community forums for further assistance. They may be able to provide additional insights or workaround solutions.

**Comment by jianmincat** [comment]
it is strange that I got below error message. But I think I have education licence for copilot?

2024-06-05 20:17:43.404 [error] [auth] Extension activation failed: "No access to GitHub Copilot found. You are currently logged in as jianmincat."

**Reply by shaymaridene** [reply]
same problem

**Reply by MohamedBoukthir** [reply]
Ensure you have the latest version of VSCode and update GitHub Copilot from the Extensions section, then enable it. After updating, reload or restart VSCode.
To check the GitHub Copilot log, go to View -> Output and select GitHub Copilot from the dropdown menu on the left.
If you're behind a proxy, verify the configuration in File -> Preferences -> Settings, filtering with 'proxy' to check Http: Proxy. If needed, click "Edit settings.json" and add:
`
"http.proxySupport": "on",
"http.proxy": "http://my-proxy.net:port"
`
Restart VSCode to apply these changes. If issues persist, please share the logs for further assistance.

**Comment by MohamedBoukthir** [comment]
1 - Update VSCode to the Latest Version:
Ensure that you are using the most recent version of VSCode for optimal performance and compatibility.

2 - Update GitHub Copilot to the Latest Version:
Go to the Extensions section in VSCode and check for updates to GitHub Copilot. Make sure it is enabled after updating.

3 - Reload/Restart VSCode:
After updating, you might need to reload or restart VSCode to apply the changes.

4 - Check the GitHub Copilot Log:
From the Menubar, navigate to View -> Output.

In the Output panel, select GitHub Copilot from the dropdown menu on the left. This will show you the log and help you understand what is happening in the background.

**Comment by jianmincat** [comment]
The issue actually resolved. I didn't confirm to receive github education benefit for copilot before. Thank you.

**Comment by chrisafl** [comment]
I am stuck on step 2, it is saying my account does not have access to copilot. I am not sure what to do from here.

![image](https://github.com/community/community/assets/7001706/2331c594-91e9-4b59-85fb-245ba260cfb7)

2024-06-14 12:32:19.130 [info] [auth] Invalid copilot token: missing token: 403 
2024-06-14 12:32:19.274 [error] [auth] Extension activation failed: "No access to GitHub Copilot found. You are currently logged in as chrisafl."

**Reply by chrisafl** [reply]
ok looking around I see that there is a signup page you have to go to. For anybody else with this issue, the tutorial doesn't mention it but you need to go here first and complete signup:

https://github.com/github-copilot/free_signup

Edit to add, I also cannot get the Codebase VSCode in the browser to work with Firefox, it is only working with Chrome/Edge for me.

**Comment by 0SOLO1** [comment]
Response #1
